crypto/core_namemap.c

```diff
@@ -216,7 +229,7 @@ int ossl_namemap_add_name_n(OSSL_NAMEMAP *namemap, int number,
         goto err;
 
     namenum->number = tmp_number =
-        number != 0 ? number : ++namemap->max_number;
+        number != 0 ? number : 1 + tsan_counter(&namemap->max_number);
```

Comment:
Side note: I had to look up what `tsan_counter` does...  the name is not at all obvious!  Could it be renamed to `tsan_incr`?  In another PR, of course.

Reply:
Yes, only one increment is possible.  Pre-increment isn't available at this point, we rely on the compiler recognising it.

Reply:
Uhmm, I only asked for a rename.  I understand that it's going to be behave like a post-increment, just like `tsan_decr` behaves like a post-decrement.

Reply:
Another PR for the rename -- perhaps as part of the rationalisation and wrapper I'd like to get done.